### PR TITLE
VER-2032: Integrate PAYE questions & validation to IVQ

### PR DESCRIPTION
**Pull request by YuanZHANG910**

### CODE REVIEW

File: app/uk/gov/hmrc/identityverificationquestions/sources/empRef/EmpRefService.scala [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import uk.gov.hmrc.identityverificationquestions.monitoring.auditing.AuditService
+import uk.gov.hmrc.identityverificationquestions.services.utilities.{CheckAvailability, CircuitBreakerConfiguration, PenceAnswerConvertor, TaxYearBuilder}
+import uk.gov.hmrc.identityverificationquestions.sources.QuestionServiceMeoMinimumNumberOfQuestions
+
+import javax.inject.{Inject, Singleton}
+
+@Singleton
+class EmpRefService @Inject()(empRefConnector: EmpRefConnector, val eventDispatcher: EventDispatcher, val auditService: AuditService)(implicit override val appConfig: AppConfig)
+  extends QuestionServiceMeoMinimumNumberOfQuestions
+    with CheckAvailability with CircuitBreakerConfiguration with TaxYearBuilder with PenceAnswerConvertor {
```
Comment (drainhmrc) [inline app/uk/gov/hmrc/identityverificationquestions/sources/empRef/EmpRefService.scala:32]: We should have a it:test for the circuit breaker logic

File: app/uk/gov/hmrc/identityverificationquestions/sources/empRef/EmpRefConnector.scala [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      val desHeaders: HeaderCarrier = headersForDES
+      val headers = desHeaders.headers(List("Authorization", "X-Request-Id")) ++ desHeaders.extraHeaders
+
+      def lastTwoYearsOfPayments(payments: List[PayePayment]): PayePaymentsDetails =
+        PayePaymentsDetails(Some(payments.filter( payment =>
+          LocalDate.parse(payment.paymentDate)
+            .isAfter(LocalDate.now().minusYears(2)))))
+
+      http.GET[PayePaymentsDetails](url, headers = headers)(implicitly, hc, ec).map { allPayePaymentsDetails =>
+        val lastTwoYearsPayments = allPayePaymentsDetails.payments.getOrElse(List()) match {
```
Comment (drainhmrc) [inline app/uk/gov/hmrc/identityverificationquestions/sources/empRef/EmpRefConnector.scala:55]: We should have unit tests covering the business logic of the payment details.

File: app/uk/gov/hmrc/identityverificationquestions/sources/empRef/EmpRefService.scala (hunk 1 above)
Comment (YuanZHANG910) [reply app/uk/gov/hmrc/identityverificationquestions/sources/empRef/EmpRefService.scala:32]: the circuit breaker test in super class

File: app/uk/gov/hmrc/identityverificationquestions/sources/empRef/EmpRefConnector.scala (hunk 2 above)
Comment (YuanZHANG910) [reply app/uk/gov/hmrc/identityverificationquestions/sources/empRef/EmpRefConnector.scala:50]: added

**Review by drainhmrc — APPROVED** [review]
👍

**Pull request merged**